src/collector/block/accountTx.ts

```diff
@@ -248,12 +210,27 @@ export default function getAddressFromMsg(
         }
       )
       break
-
-    default:
-      result = {}
-      break
   }
 
+  result.contract = ((log && log.events) || [])
+    .map((ev) => {
+      if (
+        [
+          'store_code',
+          'instantiate_contract',
+          'execute_contract',
+          'migrate_contract',
+          'update_contract_admin',
+          'clear_contract_admin',
+          'update_contract_owner'
+        ].includes(ev.type)
+      ) {
+        return ev.attributes.filter((attr) => TERRA_ACCOUNT_REGEX.test(attr.value)).map((attr) => attr.value)
+      }
```

Comment:
This map is having a side-effect in the finally result, because you are only adding the `attributes` when the type is one of the defined. So you are probably getting some like:

```js
[1, 2, 3, 4, 5, 6].map((value) => { if (value < 4) { return value } })
> [ 1, 2, 3, undefined, undefined, undefined ]
```

I will recommend use a `forEach` and iterated just one time:

```js
result.contract = []

(log?.events ?? []).forEach(({ type, attributes = [] }) => {
  if (
    [
      'store_code',
      'instantiate_contract',
      'execute_contract',
      'migrate_contract',
      'update_contract_admin',
      'clear_contract_admin',
      'update_contract_owner'
    ].includes(type)
  ) {
    attributes.forEach(({ value }) => {
      if (TERRA_ACCOUNT_REGEX.test(value)) {
        result.contract.push(value)
      }
    })
  }
})
```
This remove the need of `flat()`, `.filter(Boolean) as string[]` and filter the `attributes`. You will only iterate the logs events and attributes one time.

Reply:
I'd rather use map + flat than forEach. The reason is forEach method requires to review whole code block. I think flat, filter is more descriptive.

Reply:
```js
const a = [[1,2,3,4],[undefined,undefined,2,3,4,5]]
const b = a.flat().filter(Boolean)

// vs.

let c = []
a.forEach(k => k && k.forEach(j => j && c.push(j)))